api/types/kubernetes.go

```diff
@@ -117,7 +117,7 @@ func NewKubernetesClusterV3FromLegacyCluster(namespace string, cluster *Kubernet
 // but without secrets/credentials.
 func NewKubernetesClusterV3WithoutSecrets(cluster KubeCluster) (*KubernetesClusterV3, error) {
 	// Force a copy of the cluster to deep copy the Metadata fields.
-	copiedCluster := cluster.Copy()
+	copiedCluster := cluster.Copy().(*KubernetesClusterV3)
```

Comment:
I can pass a valid `KubeCluster` that makes this panic, with the change to `Copy`. Why not leave `Copy` like it was, and require every valid `KubeCluster` to be copyable as a `*KubernetesClusterV3` instead? I'd rather see concrete types wherever possible, so this also applies to `WindowsDesktopV3` and `DynamicWindowsDesktopV1` even though we're not asserting on those currently.